Tests/MongoSwiftSyncTests/SpecTestRunner/SpecTest.swift

```diff
@@ -215,14 +223,16 @@ extension SpecTestFile {
 
             // Due to strange behavior in mongos, a "distinct" command needs to be run against each mongos
             // before the tests run to prevent certain errors from ocurring. (SERVER-39704)
-            if MongoSwiftTestCase.topologyType == .sharded, let collName = self.collectionName {
-                for connStr in MongoSwiftTestCase.getConnectionStringPerHost() {
-                    let client = try MongoClient.makeTestClient(connStr)
-                    _ = try client.db(self.databaseName).collection(collName).distinct(fieldName: "_id")
+            if MongoSwiftTestCase.topologyType == .sharded,
+                let collName = self.collectionName,
+                test.description.contains("distinct") {
```

Comment:
this is new. why does this only apply to test cases where the description contains "distinct"? is that just the only command that suffers from the linked SERVER issue?

Reply:
Yep, see ["Why do tests that run distinct sometimes fail with StaleDbVersion"](https://github.com/mongodb/specifications/blob/master/source/transactions/tests/README.rst#why-do-tests-that-run-distinct-sometimes-fail-with-staledbversion). Originally, we just ran it before every single test run for simplicity, but it turns out this loop was taking a long time because of all the client creation and destruction. Now that we can do `runCommand` per server and don't need to construct new clients each time, it's less important that we only do this for a subset of tests, but it's an easy enough optimization that we might as well do it.